### PR TITLE
Add e2e tests for Anexia

### PR DESCRIPTION
**Pull request by ahmedwaleedmalik**
Signed-off-by: Waleed Malik <ahmedwaleedmalik@gmail.com>

**What does this PR do / Why do we need it**:
Setting an Anexia cluster manually is slightly cumbersome. To mitigate that and to be consistent with our overall testing policy for cloud providers. We have added E2E tests for Anexia as well.

**Does this PR close any issues?**:
Fixes #

**Special notes for your reviewer**:

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: cmd/conformance-tester/pkg/scenarios/anexia.go [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+		},
+	}, nil
+}
+
+func (s *anexiaScenario) MachineDeployments(_ context.Context, num int, secrets types.Secrets, cluster *kubermaticv1.Cluster) ([]clusterv1alpha1.MachineDeployment, error) {
+	// See alibaba provider for more info on this.
+	return nil, errors.New("not implemented for gitops yet")
+
+	//nolint:govet
+	md, err := createMachineDeployment(num, s.version, getOSNameFromSpec(s.osSpec), s.osSpec, providerconfig.CloudProviderAlibaba, anexiatypes.RawConfig{
```
Comment (xrstf) [inline cmd/conformance-tester/pkg/scenarios/anexia.go]: This still says Alibaba, slight copy-paste mistake ^^

**Review by xrstf — APPROVED** [review]
One minor typo, otherwise lgtm

**Review by xrstf — CHANGES_REQUESTED** [review]
Whoops, clicked the wrong option previously.

**Comment by xrstf** [comment]
So... I don't think we should merge this, if the new tests do not pass?

**Comment by ahmedwaleedmalik** [comment]
>  So... I don't think we should merge this, if the new tests do not pass?

They won't pass since we don't have working anexia setup. Please read the discussion in `sig-cluster-management`.

**Comment by ahmedwaleedmalik** [comment]
Update: CCM fails, which should be fixed with https://github.com/kubermatic/kubermatic/pull/9335. Other than that we have a working cluster, nodes, etc.

**Comment by ahmedwaleedmalik** [comment]
@xrstf ready for review 😁

**Review by xrstf — APPROVED** [review]
Love it!

/approve

**Pull request merged**